File: exercises/leetcode/704-binary-search/704-binary-search.go
// https://leetcode.com/problems/binary-search/description/

package main

func search(nums []int, target int) int {
	low := 0
	high := len(nums) - 1

	for low <= high {
		medium := (low + high) / 2
		value := nums[medium]

		if value == target {
			return medium
		}

		if value > target {
			high = medium - 1
		} else {
			low = medium + 1
		}
	}

	return -1
}

File: exercises/leetcode/704-binary-search/704-binary-search.ts
// https://leetcode.com/problems/binary-search/description/

function search (list: number[], target: number): number | null {
	let low = 0
	let high = list.length - 1
  
	while (low <= high) {
	  const medium = Math.floor((low + high) / 2)
	  const value = list[Math.round(medium)]
  
	  if (value === target) {
		return medium
	  }
  
	  if (value > target) {
		high = medium - 1
	  } else {
		low = medium + 1
	  }
	}
	
	return -1
  }